### DISCUSSION TITLE
Please Help: With a AutoMM, what is the DATA-MODEL mapping logic given multiple texts columns and multiple 'hf_text' models?

### DISCUSSION THREAD

**Discussion by hohoCode** [autogluon/autogluon#3852] [Q&A] [answered]
Say I have 2 hf_text models in a AutoMM like the below:
`model.names": ["hf_text_0", "hf_text_1", "fusion_mlp"]`

Now I have a dataframe with 3 textual columns and 1 label column, do we know which data-model mapping logic below is correct?

1. "hf_text_0" is applied to the FIRST text column; then "hf_text_1" works with the LAST TWO columns, Or,
2. "hf_text_0" is applied to the FIRST TWO text columns; then "hf_text_1" works with the LAST ONE column, Or,
3. Only "hf_text_0" gets applied to all 3 columns, "hf_text_1" got discarded. Or,
4. None of the above, please specify the correct logic.

Also, 
- What if I just have 1 text df column, what will happen with the use of 2 hf_text models?
- And what if 2 text df columns with the use of 3 hf_text models?

Please help. Any hints or codes in the repo on the DATA-Model mapping logic will help. 

Thanks a ton! 

@Innixma

**Comment by nimasteryang** [comment, marked as answer]
I assume that both model(hf_text_0 and hf_text_1) will apply to the text sequence [Col_1+[SEP]+Col_2+[SEP]+Col_3] and then concat the output of the two model before the MLP layer.
Correct me if I'm wrong.

**Reply by hohoCode** [reply]
Thanks Yang! 

In this case, if both hf_text_0 and hf_text_1 apply to same texts, then this makes this functionality not really useful. Hopefully we can have a logic to specify hf_text_0  -> text_col0 and hf_text_1  -> text_col1 .

**Reply by nimasteryang** [reply]
Agree, I believe it was designed for multimodal like text+ image. However, since Autogluon can automatically detect columns as text or images and assign them with different models, I believe it is feasible to add functionality that allow users to assign specific model for specific column/columns.

**Reply by hohoCode** [reply]
We are totally on the same page. Liked! "it is feasible to add functionality that allow users to assign specific model for specific column/columns."

@zhiqiangdon
